File: RL/Part-I/src/Event.py
class Event:
    def __init__(self, agent, pevents, index, name, site, startTime, startTimeEnd):
        self.agent = agent
        self.pevents = pevents
        self.index = index
        self.name = name
        self.site = site
        self.startTime = startTime
        self.startTimeEnd = startTimeEnd

    def __repr__(self):
        return "E: ( " + str(self.agent) + " " + str(self.pevents) + " " + str(self.name) + " " + str(self.site) +  " " + str(self.startTime) + " " + str(self.startTimeEnd) + " )"

    def __eq__(self, other):
        if isinstance(other, self.__class__):
            return self.agent == other.agent and self.index==other.index and self.name==other.name and self.site==other.site and self.startTime==other.startTime and self.startTimeEnd==other.startTimeEnd
        return False

File: EM and NLP/src/main/java/Event.java
import java.util.ArrayList;

/**
 * Created by tarun on 5/6/17.
 */
public class Event {
    int agent;
    ArrayList<PrimtiveEvent> pevents;
    int index;
    String name;
    int site;

    public Event(int agent, ArrayList<PrimtiveEvent> pevents, int index, String name, int site) {
        this.agent = agent;
        this.pevents = pevents;
        this.index = index;
        this.name = name;
        this.site = site;
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof Event)) {
            return false;
        }
        Event other = (Event) o;

        return this.index==other.index &&
                this.agent==other.agent &&
                this.name.equals(other.name)&&
                this.site==other.site;
    }

    @Override
    public String toString() {
        return "E: ( " + agent + " " + pevents.toString() + " " + name + " " + site + " )";
    }
}

